core/src/test/java/org/opensearch/sql/expression/datetime/DateTimeFunctionTest.java

```diff
@@ -1143,36 +1248,36 @@ public void testWeekOfYearDifferentModes() {
     lenient().when(missingRef.valueOf(env)).thenReturn(missingValue());
 
     //Test the behavior of different modes passed into the 'week_of_year' function
-    testWeekOfYear("2019-01-05", 0, 0);
-    testWeekOfYear("2019-01-05", 1, 1);
-    testWeekOfYear("2019-01-05", 2, 52);
-    testWeekOfYear("2019-01-05", 3, 1);
-    testWeekOfYear("2019-01-05", 4, 1);
-    testWeekOfYear("2019-01-05", 5, 0);
-    testWeekOfYear("2019-01-05", 6, 1);
-    testWeekOfYear("2019-01-05", 7, 53);
-
-    testWeekOfYear("2019-01-06", 0, 1);
-    testWeekOfYear("2019-01-06", 1, 1);
-    testWeekOfYear("2019-01-06", 2, 1);
-    testWeekOfYear("2019-01-06", 3, 1);
-    testWeekOfYear("2019-01-06", 4, 2);
-    testWeekOfYear("2019-01-06", 5, 0);
-    testWeekOfYear("2019-01-06", 6, 2);
-    testWeekOfYear("2019-01-06", 7, 53);
-
-    testWeekOfYear("2019-01-07", 0, 1);
-    testWeekOfYear("2019-01-07", 1, 2);
-    testWeekOfYear("2019-01-07", 2, 1);
-    testWeekOfYear("2019-01-07", 3, 2);
-    testWeekOfYear("2019-01-07", 4, 2);
-    testWeekOfYear("2019-01-07", 5, 1);
-    testWeekOfYear("2019-01-07", 6, 2);
-    testWeekOfYear("2019-01-07", 7, 1);
-
-    testWeekOfYear("2000-01-01", 0, 0);
-    testWeekOfYear("2000-01-01", 2, 52);
-    testWeekOfYear("1999-12-31", 0, 52);
+    weekOfYearQuery("2019-01-05", 0, 0);
```

Comment:
The same as above - is it possible to split test cases and use parameterized test?

Reply:
Refactored and added tests in https://github.com/Bit-Quill/opensearch-project-sql/pull/199/commits/3c843ba8cff46cca9ff731b204245ef8adab58bc, https://github.com/Bit-Quill/opensearch-project-sql/pull/199/commits/c664af7ba579b2eba3147ff9b05e2a6971ae8cd1, and https://github.com/Bit-Quill/opensearch-project-sql/pull/199/commits/dcbe2e22598d5404f4dee2e606ce5228aa1b7a9f